spec/compiler.js

```diff
@@ -108,4 +108,29 @@ describe('compiler', function() {
       equal(/return ""/.test(Handlebars.precompile('')), true);
     });
   });
+
+  describe('Compile Partials Correctly', function() {
+    var compileCount = 0;
+    var origTemplateFunc = Handlebars.template;
+    before(function() {
+      Handlebars.template = function(params) {
```

Comment:
I have added chai and sinon to the dev-dependencies on the current 4.x. 

Please rebase to the lastest 4.x. you will then be able to use a [sinon-spy](https://sinonjs.org/releases/latest/spies/) to count the template invocations.

I would create the spy in the `it()`-block and call `sinon.restore()` in an  `afterEach()` block.